doc/devElements.md

```diff
@@ -58,3 +58,40 @@ Function | Return object | modifiable `data` keys | unmodifiable `data` keys | D
 `grade()` | `data` (dict) | `params`, `correct_answers`, `submitted_answers`, `format_errors`, `partial_scores`, `score`, `feedback` | `variant_seed`, `options`, `raw_submitted_answers` | Grade `data["submitted_answers"][var]` to determine a score. Store the score and any feedback in `data["partial_scores"][var]["score"]` and `data["partial_scores"][var]["feedback"]`. Return the modified `data` dictionary.
 
 The above function descriptions describe the typical variables that will be read and modified by each function. However, any function that returns `data` (i.e., not `parse()`) is allowed to change any of the modifiable values in `data` (see above table) and these changes will be persisted to the database. No function is allowed to add or delete keys in `data`.
+
+### Element dependencies
+
+It's likely that your element will depend on certain client-side assets,
+such as scripts or stylesheets. To keep clean separation of HTML, CSS, and JS,
+you can place those dependencies in other files. If you depend on libraries like
+lodash or d3, you can also express that need. PrairieLearn will compile a list
+of all dependencies needed by all elements on a page, dedup the dependencies,
+and ensure they are loaded on the page.
+
+If you place `[element_name].js` or `[element_name].css` files alongside your
+Python and mustache files, they will automatically be included on the page. For
+instance, if I have an element with the tag `<pl_my_element>`, simply creating
+the files `pl_my_element.js` and `pl_my_element.css` will ensure they are loaded
+on the page.
+
+You can also manually list any scripts or stylesheets that your element relies
+on. If you want to express a dependency on a global script (anything located in
+public/javascripts/), you can include that in your dependency list as well.
+
+Dependencies are listed in `question-servers/elements/index.js`. You can
```

Comment:
Can we shift this into the element itself? This could be a function in the element code that returns a list, or it could be a `pl_my_element.json` config file. This way, if we have a course-specific element that someone wants to copy to another course, or that we want to copy into the core PL element directory, then we just have to copy all the `pl_my_element.*` files and it will just work.